101-azurefirewall-create/azuredeploy.json

```diff
@@ -49,18 +49,24 @@
       "metadata": {
         "description": "Public IP address name"
       }
-    }  
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
   },
   "variables": {
-    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks',parameters('virtualNetworkName'))]",
     "azureFirewallSubnetName": "AzureFirewallSubnet",
-    "azureFirewallSubnetId": "[concat(variables('vnetId'), '/subnets/', variables('azureFirewallSubnetName'))]",
+    "azureFirewallSubnetId":  "[resourceId('Microsoft.Network/virtualNetworks/subnets',parameters('virtualNetworkName'), variables('azureFirewallSubnetName'))]",
     "azureFirewallPublicIpId": "[resourceId('Microsoft.Network/publicIPAddresses',parameters('publicIPName'))]"
   },
   "resources": [
     {
       "name": "[parameters('virtualNetworkName')]",
-      "apiVersion": "2015-06-15",
+      "apiVersion": "2017-06-01",
       "type": "Microsoft.Network/virtualNetworks",
       "location": "[resourceGroup().location]",
```

Comment:
all of the location properties need to use the location parameter...

Reply:
Done